hybrasyl/Messaging/Admin.cs

```diff
@@ -0,0 +1,410 @@
+﻿using Hybrasyl.Creatures;
+using Hybrasyl.Objects;
+using Hybrasyl.Scripting;
+using System;
+using System.Net;
+
+namespace Hybrasyl.Messaging
+{
+    // Various admin commands are implemented here.
+
+    class SummonCommand : ChatCommand
+    {
+        public new static string Command = "summon";
+        public new static string ArgumentText = "<string playerName>";
+        public new static string HelpText = "Summon a specified player to you";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            if (!Game.World.WorldData.ContainsKey<User>(args[0]))
+                return Fail($"User {args[0]} not logged in");
+
+            var target = Game.World.WorldData.Get<User>(args[0]);
+
+            if (target.IsExempt)
+                return Fail($"User {user.Name} is exempt from your meddling");
+
+            target.Teleport(user.Map.Id, user.MapX, user.MapY);
+            return Fail("No such status was found (Missing XML file?)");
+        }
+    }
+
+    class KickCommand : ChatCommand
+    {
+        public new static string Command = "kick";
+        public new static string ArgumentText = "<string playerName>";
+        public new static string HelpText = "Kick a specified player off the server";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            if (!Game.World.WorldData.ContainsKey<User>(args[0]))
+                return Fail($"User {args[0]} not logged in");
+
+            var target = Game.World.WorldData.Get<User>(args[0]);
+
+            if (target.IsExempt)
+                return Fail($"User {target.Name} is exempt from your meddling");
+            else
+                target.Logoff();
+
+            target.Teleport(user.Map.Id, user.MapX, user.MapY);
```

Comment:
If we are kicking a player off the server, why would we be teleporting them?

Reply:
Looks like a copied method.

Reply:
yep, fixed